Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -265,6 +265,46 @@ class StoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
         expect(eligibility) == .eligible
     }
 
+    func testIneligibleForIntroForDifferentProductInSameSubscriptionGroupAfterPurchase() async throws {
+        try XCTSkipIf(
+            Self.storeKit2Setting == .enabledForCompatibleDevices,
+            "This test currently does not pass with SK2 (see FB11889732)"
+        )
+
+        let productWithNoTrial = try await self.product(Self.group3MonthlyNoTrialProductID)
+        let productWithTrial = try await self.product(Self.group3MonthlyTrialProductID)
+
+        _ = try await Purchases.shared.purchase(product: productWithNoTrial)
+
+        let eligibility = await Purchases.shared.checkTrialOrIntroDiscountEligibility(product: productWithTrial)
+        expect(eligibility) == .ineligible
+    }
+
+    func testEligibleForIntroForDifferentProductInSameSubscriptionGroupAfterExpiration() async throws {
+        let productWithNoTrial = try await self.product(Self.group3MonthlyNoTrialProductID)
+        let productWithTrial = try await self.annualPackage.storeProduct
+
+        let customerInfo = try await Purchases.shared.purchase(product: productWithNoTrial).customerInfo
```

Comment:
for some reason I never realized that you can do this syntax (i.e.: `customerInfo = try await purchase().customerinfo`. That's awesome